packages/amplify-cli/src/lib/migrate-project.js

```diff
@@ -0,0 +1,188 @@
+  delete newProjectConfig.projectPath;
+  delete newProjectConfig.defaultEditor;
+
+  // Modify frontend handler
+  const frontendPluginPath = Object.keys(projectConfig.frontendHandler)[0];
+  const frontendPlugin = frontendPluginPath.split('/')[frontendPluginPath.split('/').length - 1];
+  const frontend = frontendPlugin.split('-')[frontendPlugin.split('-').length - 1];
+
+  newProjectConfig.frontend = frontend;
+  delete newProjectConfig.frontendHandler;
+
+  // Modify provider handler
+  const provider = Object.keys(projectConfig.providers)[0];
+  newProjectConfig.providers = [provider];
+
+  const jsonString = JSON.stringify(newProjectConfig, null, 4);
+  fs.writeFileSync(projectConfigFilePath, jsonString, 'utf8');
+}
+
+function generateLocalEnvInfo(projectConfig) {
+  const envInfo = {
+    projectPath: projectConfig.projectPath,
+    defaultEditor: projectConfig.defaultEditor,
+    envName: 'NONE',
+  };
+
+  const jsonString = JSON.stringify(envInfo, null, 4);
+  const localEnvFilePath = getLocalEnvFilePath();
+  fs.writeFileSync(localEnvFilePath, jsonString, 'utf8');
+}
+
+function generateAwsLocalInfo() {
+  const dotConfigDirPath = getDotConfigDirPath();
+  const awsInfoFilePath = path.join(dotConfigDirPath, 'aws-info.json');
+  if (fs.existsSync(awsInfoFilePath)) {
+    const awsInfo = JSON.parse(fs.readFileSync(awsInfoFilePath));
+    awsInfo.configLevel = 'project'; // Old version didn't support "General" configuation
+    const newAwsInfo = { NONE: awsInfo };
+
+    const jsonString = JSON.stringify(newAwsInfo, null, 4);
+    const localAwsInfoFilePath = path.join(dotConfigDirPath, 'local-aws-info.json');
+    fs.writeFileSync(localAwsInfoFilePath, jsonString, 'utf8');
+    fs.removeSync(awsInfoFilePath);
+  }
+}
+
+function generateTeamProviderInfo() {
+  const amplifyMetafilePath = getAmplifyMetaFilePath();
+  const amplifyMeta = JSON.parse(fs.readFileSync(amplifyMetafilePath));
+
+  const teamProviderInfo = { NONE: amplifyMeta.providers };
+
+  const jsonString = JSON.stringify(teamProviderInfo, null, 4);
+  const teamProviderFilePath = getProviderInfoFilePath();
+  fs.writeFileSync(teamProviderFilePath, jsonString, 'utf8');
+}
+
+function generateBackendConfig() {
+  const amplifyMetafilePath = getAmplifyMetaFilePath();
+  const amplifyMeta = JSON.parse(fs.readFileSync(amplifyMetafilePath));
+
+  const backendConfig = {};
+  Object.keys(amplifyMeta).forEach((category) => {
+    if (category !== 'providers') {
+      backendConfig[category] = {};
+      Object.keys(amplifyMeta[category]).forEach((resourceName) => {
+        backendConfig[category][resourceName] = {};
+        backendConfig[category][resourceName].service = amplifyMeta[category][resourceName].service;
+        backendConfig[category][resourceName].providerPlugin =
+        amplifyMeta[category][resourceName].providerPlugin;
+      });
+    }
+  });
+
+
+  const jsonString = JSON.stringify(backendConfig, null, 4);
```

Comment:
I see these repeated, can we move this a function. I know its 2 lines but still its worth moving it in case if we have to add more checks

Reply:
Yeah, we can. As you mentioned that this is just two lines, but don't have a strong opinion on this one.

Reply:
Adding a TODO out here: